### PR TITLE
Accessibility requirements mapping rewrite

### PR DESCRIPTION
**Pull request by WilcoFiers**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wcag-act/pull/313.html" title="Last updated on Jan 13, 2019, 1:17 PM UTC (ede6f60)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wcag-act/313/6ea8ddc...ede6f60.html" title="Last updated on Jan 13, 2019, 1:17 PM UTC (ede6f60)">Diff</a>

### CODE REVIEW

**Comment by tbostic32** [comment]
I really like the updates, I think it really spells out how the requirements intertwine with the rules. I have a couple changes/comments:
The word "required" is misspelled in example 6 and 7. 

> If the failed outcome can not be mapped to not satisfied for an accessibility requirement, that requirement MUST NOT be listed under the accessibility requirements mapping.

What is the explanation for including the failed outcome if it will always be mapped to the same value? I think the reasoning should be included in the document.

File: act-rules-format.bs [hunk 1]
```diff
@@ -110,42 +110,88 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+        <li><strong>Required for conformance</strong> to WCAG 2.0 and WCAG 2.1 level A</li>
+        <li>Outcome mapping: <ul>
+          <li>Failed outcome: not satisfied</li>
+          <li>Passed outcome: further testing is needed</li>
+          <li>Inapplicable outcome: further testing is needed</li>
+        </ul></li>
+      </ul>
+    </li>
+    <li><a href="https://www.w3.org/TR/WCAG21/#name-role-value">
+        WCAG 2.1, Level A, Success Criterion 4.1.2: Name, Role, Value
```
Comment (WilcoFiers) [inline act-rules-format.bs]: Should I remove the level here?

File: act-rules-format.bs [hunk 2]
```diff
@@ -110,42 +110,88 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+          <li>Passed outcome: further testing is needed</li>
+          <li>Inapplicable outcome: further testing is needed</li>
+        </ul></li>
+      </ul>
+    </li>
+  </ul></blockquote>
+</div>
 
-If any accessibility requirements are listed, the ACT Rule MUST indicate how each outcome impacts the conformance to the listed accessibility requirements. For WCAG 2.1 Success Criteria, this means indicating whether or not the success criterion can be satisfied, or if further testing is needed.
+ACT Rules can be used to test accessibility requirements that are not part of a W3C accessibility standard, such as WCAG advisory techniques, or a testing methodology like RGAA. When an ACT Rule is used to test something not required by its accompanying standard, the rule MUST indicate this as part of its accessibility requirements mapping. Examples of this are WCAG sufficient techniques, which are not required for WCAG conformance, or a company style guide which includes both requirements and optional "best practices". That distinction between what is required what is optional has to be made clear.
```
Comment (WilcoFiers) [inline act-rules-format.bs]: Should say "what is required **and** what is optional".

**Comment by moekraft** [comment]
Some editorial comments. 
1. Should "must" be capitalized? In the first paragraph it is lowercase, "An accessibility  requirement that  is  a particular web page conforms to for it  requirement related  to accessibility, that must  be considered accessible. satisfied in order to conform to a standard, or to comply with a contract, policy or regulation."
2. We are not consistent with case for "failed". Sometimes we capitalize the first letter.
3. I think we should we add a reference link to Outcome when we first starting mentioning outcome failed. I think this could be a separate issue.

File: act-rules-format.bs [hunk 3]
```diff
@@ -110,42 +110,88 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+  <p><strong>Example:</strong>Accessibility Requirements Mapping for a rule that tests that each `img` element has an `alt` attribute</p>
+  <blockquote><ul>
+    <li><a href="https://www.w3.org/TR/WCAG20-TECHS/H37.html">
+      Technique H37: Using alt attributes on img elements
+      </a><ul>
+        <li><strong>Not required</strong> for conformance to WCAG 2.0 and WCAG 2.1 at any level</li>
+        <li>Outcome mapping: <ul>
+          <li>Failed outcome: not satisfied</li>
+          <li>Passed outcome: satisfied</li>
+          <li>Inapplicable outcome: satisfied</li>
```
Comment (WilcoFiers) [inline act-rules-format.bs]: This probably isn't satisfied. Needs further testing.

File: act-rules-format.bs [hunk 4]
```diff
@@ -110,42 +110,88 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+      </ul>
+    </li>
+    <li><a href="https://disic.github.io/rgaa_referentiel_en/criteria.html#test-1-1-1">
+      RGAA 3, Test 1.1.1: Does each image have a text alternative
+      </a><ul>
+        <li><strong>Required for conformance</strong> to RGAA 3 level A</li>
+        <li>Outcome mapping: <ul>
+          <li>Failed outcome: not satisfied</li>
+          <li>Passed outcome: satisfied</li>
+          <li>Inapplicable outcome: satisfied</li>
```
Comment (WilcoFiers) [inline act-rules-format.bs]: This probably isn't satisfied. Needs further testing.

File: act-rules-format.bs [hunk 5]
```diff
@@ -17,7 +17,7 @@ Introduction {#intro}
 
 There are currently many test procedures and tools available which aid their users in testing web content for conformance to accessibility standards such as the [Web Content Accessibility Guidelines (WCAG)](https://www.w3.org/WAI/standards-guidelines/wcag/) [[WCAG]]. As the web develops in both size and complexity, these procedures and tools are essential for managing the accessibility of resources available on the web.
 
-This format is intended to enable a consistent interpretation of how to test for [accessibility requirements](#accessibility-requirements) and promote consistent results of accessibility tests. It is intended to be used to describe both manual accessibility tests as well as automated tests performed by accessibility testing tools.
+This format is intended to enable a consistent interpretation of how to test [=accessibility requirements documents=] and promote consistent results of accessibility tests. It is intended to be used to describe both manual accessibility tests as well as automated tests performed by accessibility testing tools.
```
Comment (annethyme) [inline act-rules-format.bs]: Should it be "how to create tests for accessibility requirements documents" or "how to test for the requirements in accessibility requirements documents"?
We are not really testing the documents, but the requirements in the documents...

File: act-rules-format.bs [hunk 6]
```diff
@@ -110,42 +110,86 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+An accessibility requirement is a requirement related to accessibility, that must be satisfied in order to conform to a standard, or to comply with a contract, policy or regulation. A common example of accessibility requirements are the WCAG 2.1 success criteria. There are other standards, including W3C standards that include recommendations for accessibility, such as WAI-ARIA and HTML. Accessibility requirements are also often found in company policies, regional standards or in legislation. An example of a legislation is the European Web Accessibility Directive.
 
+An ACT Rule SHOULD list each accessibility requirement that is not satisfied when the outcome of the rule is `failed`. Each accessibility requirement listed this way MUST include the following:
 
-Accessibility Requirements List {#accessibility-requirements-list}
-------------------------------------------------------------------
+1. the name, title or summary of the accessibility requirement, and
+2. the name of the standard or policy that includes the accessibility, and
```
[the remark this answers is not in the document]
Comment (annethyme) [reply act-rules-format.bs]: "standard or policy" --> "accessibility requirements document"

File: act-rules-format.bs [hunk 7]
```diff
@@ -110,42 +110,90 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+Each [=accessibility requirement=] in the mapping MUST include the following:
 
-Accessibility Requirements List {#accessibility-requirements-list}
-------------------------------------------------------------------
+1. the name, title or summary of the accessibility requirement, and
+2. the name of the standard or policy that includes the accessibility, and
+3. the conformance level associated with the accessibility requirement, if one exists.
 
-An ACT Rule SHOULD list an identifier and reference for each accessibility requirement that it maps to.
+For each accessibility requirement in the mapping, an ACT Rule MUST indicate what the outcome means for satisfying that accessibility requirement. When the outcome of a rule is `failed`, the accessibility requirements are *<dfn>not satisfied</dfn>*. When the outcome of a rule is `passed` or `inapplicable`, the accessibility requirements could be *<dfn>satisfied</dfn>*, or *<dfn>further testing could be necessary</dfn>*. Rules that can be used to determine if an accessibility requirement is *satisfied* are called <dfn>satisfying tests</dfn>.
```
Comment (annethyme) [inline act-rules-format.bs]: "outcome" --> "outcome of the rule"?

File: act-rules-format.bs [hunk 8]
```diff
@@ -110,42 +110,90 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
-
-Accessibility Requirements Mapping {#accessibility-requirements-mapping}
-------------------------------------------------------------------------
+<div class=example>
+  <p><strong>Example:</strong> Accessibility Requirements Mapping for a rule that tests if an image button has an accessible name:
+  <blockquote><ul>
+    <li><a href="https://www.w3.org/TR/WCAG21/#non-text-content">
+        Success Criterion 1.1.1: Non-text content
+      </a><ul>
+        <li><strong>Required for conformance</strong> to WCAG 2.0 and WCAG 2.1 level A</li>
```
Comment (annethyme) [inline act-rules-format.bs]: Is this "required/not required for conformance" even mentioned in the specification outside of this example?

Comment (annethyme) [inline act-rules-format.bs]: When listing  "the name of the standard or policy that includes the accessibility requirement", is it enough to do it as part of something else like done here?

Comment (annethyme) [reply act-rules-format.bs]: ... it is apparently mentioned below this example, which seems a bit odd...

File: act-rules-format.bs [hunk 9]
```diff
@@ -110,42 +110,90 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
-  **Note**: In WCAG 2.1, success criteria do not evaluate to `passed`, `failed` or `inapplicable`. Rather they can be *satisfied* (or not). (See the [WCAG 2.1 definition: satisfies a success criterion](https://www.w3.org/TR/WCAG21/#dfn-satisfies).) If a success criterion is *not satisfied* a web page can only conform if there is a conforming alternative version, as described in [WCAG 2.1 Conformance Requirement 1](https://www.w3.org/TR/WCAG21/#cc1).
+If the `failed` outcome can not be mapped to *not satisfied* for an accessibility requirement, that requirement MUST NOT be included in the accessibility requirements mapping. The optional Background section could be used to list accessibility requirements and standards that may be thematically related, but for which the rule is not a failure test.
+
+If the rule does not map to any [=accessibility requirement=], the accessibility requirement mapping will only contain the explainer that it is not required for conformance to the [=accessibility requirements document=]. This is common with atomic rules used in composite rules, where accessibility requirements are only not satisfied with multiple atomic rules have `failed`.
```
Comment (annethyme) [inline act-rules-format.bs]: "with multiple atomic rules have `failed`" --> "when multiple atomic rules have the outcome `failed` "

File: act-rules-format.bs [hunk 10]
```diff
@@ -110,42 +110,90 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+          <li>Inapplicable outcome: satisfied</li>
+        </ul></li>
+      </ul>
+    </li>
+  </ul></blockquote>
 </div>
 
-<div class=note>
-  **Note**: In WCAG 2.1, success criteria do not evaluate to `passed`, `failed` or `inapplicable`. Rather they can be *satisfied* (or not). (See the [WCAG 2.1 definition: satisfies a success criterion](https://www.w3.org/TR/WCAG21/#dfn-satisfies).) If a success criterion is *not satisfied* a web page can only conform if there is a conforming alternative version, as described in [WCAG 2.1 Conformance Requirement 1](https://www.w3.org/TR/WCAG21/#cc1).
+If the `failed` outcome can not be mapped to *not satisfied* for an accessibility requirement, that requirement MUST NOT be included in the accessibility requirements mapping. The optional Background section could be used to list accessibility requirements and standards that may be thematically related, but for which the rule is not a failure test.
```
Comment (annethyme) [inline act-rules-format.bs]: Should "Background" be a link?

Comment (annethyme) [reply act-rules-format.bs]: Should there be a definition for "failure test"?

File: act-rules-format.bs [hunk 11]
```diff
@@ -110,42 +110,90 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
+          <li>Passed outcome: further testing is needed</li>
+          <li>Inapplicable outcome: further testing is needed</li>
+        </ul></li>
+      </ul>
+    </li>
+  </ul></blockquote>
+</div>
 
-If any accessibility requirements are listed, the ACT Rule MUST indicate how each outcome impacts the conformance to the listed accessibility requirements. For WCAG 2.1 Success Criteria, this means indicating whether or not the success criterion can be satisfied, or if further testing is needed.
+ACT Rules can be used to test accessibility requirements that are not part of a W3C accessibility standard, such as WCAG advisory techniques, or a testing methodology like RGAA. When an ACT Rule is used to test something not required by its accompanying standard, the rule MUST indicate this as part of its accessibility requirements mapping. Examples of this are WCAG sufficient techniques, which are not required for WCAG conformance, or a company style guide which includes both requirements and optional "best practices". That distinction between what is required and what is optional has to be made clear.
```
Comment (annethyme) [inline act-rules-format.bs]: Should we have an explanation for what RGAA is? Or a link to it maybe?
And maybe the same for WCAG advisory techniques?

Comment (annethyme) [reply act-rules-format.bs]: Is it also a MUST to indicate when something IS required for conformance?

File: act-rules-format.bs (hunk 8 above)
Comment (annethyme) [reply act-rules-format.bs]: ... and should we link to the standards themselves too?

File: act-rules-format.bs (hunk 6 above)
[the remark this answers is not in the document]
Comment (annethyme) [reply act-rules-format.bs]: Could we suggest that if possible, a link to the standard is included too?

File: act-rules-format.bs (hunk 5 above)
Comment (WilcoFiers) [reply act-rules-format.bs]: Changed it to "how to test for conformance to accessibility requirements documents".

File: act-rules-format.bs (hunk 6 above)
[the remark this answers is not in the document]
Comment (WilcoFiers) [reply act-rules-format.bs]: Made all the changes.

File: act-rules-format.bs [hunk 12]
```diff
@@ -110,42 +110,95 @@ An ACT Rule MUST have a description that is in plain language, and provides a br
...
 </div>
 
 <div class=note>
-  **Note**: In WCAG 2.1, success criteria do not evaluate to `passed`, `failed` or `inapplicable`. Rather they can be *satisfied* (or not). (See the [WCAG 2.1 definition: satisfies a success criterion](https://www.w3.org/TR/WCAG21/#dfn-satisfies).) If a success criterion is *not satisfied* a web page can only conform if there is a conforming alternative version, as described in [WCAG 2.1 Conformance Requirement 1](https://www.w3.org/TR/WCAG21/#cc1).
+**Note**: While rules are often designed for one, or possibly a small collection of [=accessibility requirements document=], it is likely that other accessibility requirements documents also map to those rules. For example, rules may be written for WCAG 2.1, but many of those may map to a company's internal accessibility policy. In such a scenario, an external accessibility requirements mapping could be created. An external accessibility requirements mapping ammend the accessibility requirements mapping of a rule by adding mapping to a different accessibility requirements document. An external accessibility requirements mapping avoids duplication of a rule, for the sole purpose of changing the mapping.
```
Comment (WilcoFiers) [inline act-rules-format.bs]: This note is new. Please review.

File: act-rules-format.bs (hunk 8 above)
Comment (WilcoFiers) [reply act-rules-format.bs]: TODO: Add link to RGAA. Make sure to also mention when something is required for conformance.

**Pull request merged**